app/containers/MapContainer/utils.js

```diff
@@ -57,6 +57,105 @@ export const valueOfCircle = (radius, range, config) => {
   return scale(radius);
 };
 
+export const filterFeatureMaxZoom = (feature, zoom) => {
+  if (
+    feature.properties && feature.properties.marker_max_zoom
+  ) {
+    return zoom <= parseInt(feature.properties.marker_max_zoom, 10);
+  }
+  return false;
+};
+
+// append point countries onto map
+export const getPointLayer = ({ data, config, markerEvents }) => {
+  const layer = L.featureGroup(null);
+  const {
+    indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle, zoom,
+  } = config;
+  const events = {
+    mouseover: (e) => markerEvents.mouseover ? markerEvents.mouseover(e, config) : null,
+    mouseout: (e) => markerEvents.mouseout ? markerEvents.mouseout(e, config) : null,
+    click: (e) => (markerEvents.click ? markerEvents.click(e, config) : null),
+  };
+
+  const jsonLayer = L.geoJSON(data, {
+    pointToLayer: (feature, latlng) => {
+      const filterFeature = filterFeatureMaxZoom(feature, zoom);
+      const styles = getAreaLayer(feature, indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle);
+      return L.circleMarker(latlng, {
+        radius: 6,
+        zIndex: 1,
+        pane: 'markerPane',
+        opacity: filterFeature ? 1 : 0,
+        fillOpacity: filterFeature ? 1 : 0,
+        ...styles,
+      }).on(events);
+    },
+  });
+
+  layer.addLayer(jsonLayer);
+
+  return layer;
+};
+
+// generate styles for a feature
+export const getAreaLayer = (feature, indicator, mapOptions, mapSubject, maxValueCountries, styleType, valueToStyle) => {
```

Comment:
if kept (probably not needed) should renamed to something like `getAreaLayerStyle` (as we are not returning a Leaflet layer object here)